codejail_service/startup_check.py

```diff
@@ -131,3 +144,36 @@ def _check_escape_subprocess():
         return f"Expected permission or resource limit error, but got: {error_message}"
 
     return True
+
+
+def _check_network_access():
+    """
+    Check for denial of network access.
+    """
+    # Even creating a socket should fail (before we even try to connect or
+    # bind). Because this operation doesn't actually interact with the network,
+    # it should avoid misleading errors while running unit tests under
+    # unfavorable network conditions, and doesn't require any consideration of
+    # properly tuned timeouts. (Attempting a DNS or HTTP request here would run
+    # into those problems.)
+    #
+    # We go to the trouble of properly closing the socket because there's a unit
+    # test where the socket is successfully created, and if we don't close it
```

Comment:
Maybe I'm tired. Where is this?

Reply:
It's in the `with`!

Here's what I had previously, and what provoked the failure I described:

```
import socket
filedesc = socket.socket(socket.AF_INET, socket.SOCK_STREAM).fileno()
```

And then I fixed it by explicitly calling close on the socket, and then remembered I could just use `with` like I would normally do. But the comment could be clearer about where the close happens, since it's a bit hidden here.